src/device-state.coffee

```diff
@@ -605,13 +605,13 @@ module.exports = class DeviceState extends EventEmitter
 			@applications.localModeSwitchCompletion()
 		.then =>
 			@usingInferStepsLock =>
-				@applications.getExtraStateForComparison()
-				.then (extraState) =>
-					Promise.all([
-						@getCurrentForComparison()
-						@getTarget({ initial, intermediate })
-					])
-					.then ([ currentState, targetState ]) =>
+				Promise.all([
+					@getCurrentForComparison()
+					@getTarget({ initial, intermediate })
+				])
+				.then ([ currentState, targetState ]) =>
+					@applications.getExtraStateForComparison(currentState, targetState)
```

Comment:
I'm afraid this change of ordering would re-introduce the race condition fixed in https://github.com/balena-io/balena-supervisor/commit/42737cb9e96b889d0fdfd48926778a38566d927b

Not sure what'd be the best way to fix it though - I think you can get around it by calling `@images.getAvailable` before getting the current state, and removing it from `getExtraStateForComparison`. But the "correct" fix probably involves changing ApplicationManager's `normaliseAndExtendAppFromDB` to not do calls to `@images` directly and instead take the available images as an argument.

Reply:
@CameronDiver thoughts here?

Reply:
This is fixed already (see Pablo's approval).

Reply:
Got it, thanks.